include/swift/AST/Attr.def

```diff
@@ -381,10 +381,11 @@ SIMPLE_DECL_ATTR(_usableFromInline, UsableFromInlineImport,
 // SWIFT_ENABLE_TENSORFLOW
 DECL_ATTR(differentiable, Differentiable,
           OnFunc | LongAttribute, 78)
-
 SIMPLE_DECL_ATTR(compilerEvaluable, CompilerEvaluable,
                  OnAccessor | OnFunc | OnConstructor | OnSubscript,
                  /* Not serialized */ 79)
+SIMPLE_DECL_ATTR(TensorFlowGraph, TensorFlowGraph,
```

Comment:
I'm surprised that `@convention(tensorflow)` doesn't work on declarations.  I would expect that you could use it like this:
@convention(tensorflow)
func foo() {}

@jckarter is there some other way to declare a function with a specified calling convention already?

If @convention is the wrong way to go, then this is ok with me, this means that we'll have to have two things though: @convention(tensorflow) for closures, and this attribute for decls :-(

Reply:
No, because the convention of declarations doesn't normally matter. We generate a function value with the right convention when you reference a decl as a value of `@convention(whatever)` type.

Reply:
Aha, that makes sense @jckarter and you're right that the same idea applies here.  We'll explore this and figure it out.  Thank you!

Reply:
In your case, I can see the use for a decl-side attribute, since you probably want to enforce that a `@TensorFlowGraph` function only performs operations that make sense on a TPU, which is similar to things like constexpr, "pure", "doesn't allocate", "throws", "async", "is thread safe", etc. in that it's a transitive property, and makes sense as both a decl-side restriction and a constraint on function values.